scripts/globals/abilities/full_circle.lua

```diff
@@ -0,0 +1,38 @@
+-----------------------------------
+-- Ability: Full Circle
+-- Causes your luopan to vanish.
+-- Obtained: Geomancer Level 5
+-- Recast Time: 10 seconds
+-- Refunds some of the MP consumed by the Geocolure spell that created the luopan.
+-- Amount of MP restored varies depending on remaining Luopan HP.
+-----------------------------------
+require("scripts/globals/ability")
+require("scripts/globals/status")
+require("scripts/settings/main")
+require("scripts/globals/msg")
+-----------------------------------
+local ability_object = {}
+
+ability_object.onAbilityCheck = function(player, target, ability)
+    if not player:hasPet() or not player:getPetID() == 75 then
+        return xi.msg.basic.REQUIRE_LUOPAN, 0
+    end
+    return 0,0
```

Comment:
Can we extract this block into the job utils for geo?

```lua
ability_object.onAbilityCheck = function(player, target, ability)
    return xi.job_utils.geomancer.checkLuopan(player, target, ability)
end

...

xi.job_utils.geomancer.checkLuopan = function(player, target, ability)
    if not player:hasPet() or not player:getPetID() == 75 then
        return xi.msg.basic.REQUIRE_LUOPAN, 0
    end

    return 0, 0
end
```

Reply:
this has been done now, makes more sense